sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVTypeCastSuite.scala

```diff
@@ -59,45 +49,4 @@ class CSVTypeCastSuite extends SparkFunSuite {
     }
     assert(exception.getMessage.contains("Unsupported special character for delimiter"))
   }
-
-  test("Nullable types are handled") {
```

Comment:
I think it is better to keep these tests. I suggest you add a function to `CSVRelation` companion object for type casting and then keep these unit tests.